gp-includes/template.php

```diff
@@ -258,8 +258,37 @@ function gp_project_links_from_root( $leaf_project ) {
 	return $links;
 }
 
-function gp_breadcrumb_project( $project ) {
-	return gp_breadcrumb( gp_project_links_from_root( $project ) );
+/**
+ * Get project breadcrumb.
+ *
+ * @since 4.0.0   New $extra_items array to append items like Translation Set, Project Glossary or the current action. The last item has no link.
+ *                If project ID is '0', set base locales link to breadcrumb.
+ *
+ * @param GP_Project $project       GlotPress Project object.
+ * @param array      $extra_items   Array of additional items to add to the breadcrumb.
+ *
+ * @return string   HTML of the breadcrumb.
+ */
+function gp_breadcrumb_project( $project, $extra_items = array() ) {
+
+	// If is a translation project, get the links. If is a virtual project with ID '0' for glossary, return base Locale for breadcrumb.
+	$breadcrumb = 0 !== $project->id ? gp_project_links_from_root( $project ) : array( gp_link_get( gp_url( '/languages' ), __( 'Locales', 'glotpress' ) ) );
+
+	foreach ( $breadcrumb as $key => $link ) {
+		// If no extra items, the last item is the project name without link.
+		if ( array_key_last( $breadcrumb ) === $key && empty( $extra_items ) ) {
+			$breadcrumb[ $key ] = $project->name;
+		}
+	}
```

Comment:
To avoid the loop, I suggest going to the last item in the array and update its value
```suggestion
	if ( empty( $extra_items ) ) {
		end( $breadcrumb );
		$last_key = key( $breadcrumb );
		$breadcrumb[ $last_key ] = $project->name;
	}
```